src/Windows/Avalonia.Win32/Win32Platform.cs

```diff
@@ -189,8 +189,11 @@ public void ProcessMessage()
 
             if (UnmanagedMethods.GetMessage(out var msg, IntPtr.Zero, 0, 0) > -1)
             {
-                UnmanagedMethods.TranslateMessage(ref msg);
-                UnmanagedMethods.DispatchMessage(ref msg);
+                if (!WindowImpl.PreProcessMessage(msg))
```

Comment:
I see one problem with this - my app is running a custom window loop so I might need a way of invoking Avalonia message handling somehow. Since my app is basically doing:
```
while (PeekMessage(...)) {
  TranslateMessage();
  DispatchMessage()
}
```

This allows my update loop to still run and process win32 messages when I need to.

Reply:
We're already exposing the message loop in this method, so I guess we also need to expose a public `PreProcessMessage` method here?

Reply:
That won't help if custom message loop is run by Windows itself at some point, which happens quite frequently. So I think it's better to raise a flag in WindowImpl itself and check if if the next arrived message is WM_CHAR.

Reply:
That might be better (and simpler tbh). I didn't go that way because WPF does the `PreProcessMessage` stuff, but it also has hacks where it sets a flag as well, so I'm guessing they encountered the same thing after implementing the pre-proccessing.  

I'll close this PR then and open another.